pkg/issuer/ca/ca.go

```diff
@@ -22,14 +24,19 @@ type CA struct {
 	secretsLister            corelisters.SecretLister
 }
 
-func NewCA(issuer v1alpha1.GenericIssuer,
+func NewCA(issuerObj v1alpha1.GenericIssuer,
 	cl kubernetes.Interface,
 	cmclient clientset.Interface,
 	recorder record.EventRecorder,
 	issuerResourcesNamespace string,
 	secretsLister corelisters.SecretLister) (issuer.Interface, error) {
+
+	if err := issuer.ValidateDuration(issuerObj); err != nil {
+		return nil, fmt.Errorf("CA %s", err.Error())
+	}
```

Comment:
Instead of checking ValidateDuration in the New* functions, we should perform it in Setup() and if they are invalid, mark the Issuer NotReady. This will cause Certificate requests to fail as well as a side effect, and providers a cleaner error reporting interface (we can add a status condition in future to explain why it failed, e.g. a `ConfigValid` status condition)